x-pack/platform/plugins/shared/cases/server/client/utils.ts

```diff
@@ -636,9 +645,29 @@ export const transformTemplateCustomFields = ({
     }
 
     // remove deleted custom field from template
-    const transformedTemplateCustomFields = templateCustomFields.filter((templateCustomField) =>
-      customFields?.find((customField) => customField.key === templateCustomField.key)
-    );
+    const transformedTemplateCustomFields = templateCustomFields
+      .filter((templateCustomField) =>
+        customFields?.find((customField) => customField.key === templateCustomField.key)
+      )
+      // From list fields, remove deleted options
+      .map((templateCustomField) => {
+        if (templateCustomField.type === CustomFieldTypes.LIST && templateCustomField.value) {
+          const configuration = customFields.find(
+            (customField) => customField.key === templateCustomField.key
+          ) as ListCustomFieldConfiguration;
+          const valueExistsInConfig = !!configuration.options.find(
+            ({ key }) =>
+              templateCustomField.value && Object.keys(templateCustomField.value)[0] === key
+          );
+          if (!valueExistsInConfig) {
+            return {
+              ...templateCustomField,
+              value: null,
+            };
+          }
+        }
+        return templateCustomField;
+      });
 
     // add new custom fields to template
     if (customFields.length >= transformedTemplateCustomFields.length) {
```

Comment:
We also need logic to add new list custom field's default value.  Currently if you create a `new list custom field with default value` via API it throws `invalid value` error for template.